src/adaptors/in-memory-articles.ts

```diff
@@ -10,7 +10,7 @@ export default class InMemoryArticles implements Articles {
 
   async add(article: JsonLdObj): Promise<void> {
     if (!('@type' in article) || !(article['@type'].includes(schema.Article))) {
-      throw new NotAnArticle(typeof article['@type'] === 'string' ? [article['@type']] : article['@type']);
+      throw new NotAnArticle([].concat(article['@type'] || []));
```

Comment:
Looked at using https://www.npmjs.com/package/lodash.castarray too, but also results in `[undefined]` (without `|| []`), so not worth a new library.

Reply:
In fact https://github.com/libero/article-store/blob/0ee7221bc9d90d4150abc67f87d733daa02edb2f/src/adaptors/in-memory-articles.ts#L12 requires it to be an array anyway!

Reply:
Or not quite, odd behaviour if it's a string. Refactoring.

Reply:
https://github.com/libero/article-store/pull/88/commits/8a743947821b7769303d06f978c2437173662a05